packages/ag-charts-community/src/chart/legend.ts

```diff
@@ -765,6 +769,17 @@ export class Legend {
         let newEnabled = enabled;
         if (toggleSeriesVisible) {
             newEnabled = !enabled;
+
+            if (preventHidingAll && !newEnabled) {
+                const numVisibleItems = chartService.series
+                    .flatMap((series) => series.getLegendData('category'))
+                    .filter((datum) => datum.enabled).length;
+
+                if (numVisibleItems < 2) {
+                    newEnabled = true;
+                }
+            }
+
```

Comment:
This will fire the event even if the item was already enabled, but if for whatever reason all the legend items are disabled, this will allow them to be toggled back on again.